certificates/src/lib.rs

```diff
@@ -103,15 +104,15 @@ impl CertificateValidator {
     /// `key` is expected to be the private key for the leaf certificate of
     /// `certs` in PEM format.
     ///
-    /// If `hostname` is not `None`, the leaf certificate of `certs` must be
-    /// valid for `hostname`, as determined by a dNSName entry in its subject
-    /// alternate names or (if there are no dNSName SANs) the cert's common
-    /// name.
-    pub fn validate(
+    /// If `possible_hostnames` is empty, no hostname validation is performed.
```

Comment:
Should `possible_hostnames` being empty just be an error?

Reply:
There's one path in wicket where we currently validate a cert without a hostname: if the operator uploads a cert _before_ uploading the TOML config, we don't have a hostname to validate but can still check that the key matches the cert. We will still validate the cert with the recovery silo hostname prior to kicking off RSS. We could change that - it doesn't seem unreasonable to say "you must upload the TOML before uploading certs/keys"?

Reply:
true, i think it's fine as-is then.